src/controller/CHIPDeviceController.cpp

```diff
@@ -466,6 +469,13 @@ DeviceCommissioner::DeviceCommissioner() :
     mDeviceNOCChainCallback(OnDeviceNOCChainGeneration, this), mSetUpCodePairer(this)
 {}
 
+DeviceCommissioner::~DeviceCommissioner()
+{
+#if CHIP_DEVICE_CONFIG_ENABLE_WIFIPAF
+    DeviceLayer::ConnectivityMgr().WiFiPAFCancelConnect();
```

Comment:
Cancellation like this should happen in shutdown, not in the destructor.

Reply:
https://github.com/project-chip/connectedhomeip/issues/34612